### PR TITLE
skeletonize maintenance

### PR DESCRIPTION
**Pull request by rfezzani**
## Description

This PR introduces few improvements `skeletonize`:

- code style maintenance,
- avoid data copy if not necessary,
- use `uint8` instead of `int` when possible
- reorder pixel removal test according to LUT values occurrence to save unnecessary tests

### CODE REVIEW

**Comment by alexdesiqueira** [comment]
Great stuff! Thanks @rfezzani!

**Review by alexdesiqueira — APPROVED** [review]

File: skimage/morphology/_skeletonize_cy.pyx [hunk 1]
```diff
@@ -40,32 +40,36 @@ def _fast_skeletonize(image):
...
     # this removes the need to handle border cases below
     _skeleton = np.zeros((nrows, ncols), dtype=np.uint8)
-    _skeleton[1:nrows-1, 1:ncols-1] = image
+    _skeleton[1:-1, 1:-1] = image
 
     _cleaned_skeleton = _skeleton.copy()
 
+
     # cdef'd numpy-arrays for fast, typed access
     cdef cnp.uint8_t [:, ::1] skeleton, cleaned_skeleton
```
Comment (lagru) [inline skimage/morphology/_skeletonize_cy.pyx:74]: I think we may not need these temporary variables `_skeleton` and `_cleaned_skeleton` at all. [`skeleton.base[1:-1, 1:-1]` could be used to return the actual array and not the memoryview](https://cython.readthedocs.io/en/latest/src/userguide/memoryviews.html#memoryview-objects-and-cython-arrays) at the end of this function...

**Review by lagru — APPROVED** [review]
@rfezzani thank you. I think the code is a lot less confusing now. Some of the style changes (e.g. new empty lines) I don't really see as improvements or seem strange to me (trailing `]` not on a new line). I'll wait a bit in case you want to address my suggestion otherwise I'll go ahead and merge (or someone else).

File: skimage/morphology/_skeletonize_cy.pyx (hunk 1 above)
Comment (rfezzani) [reply skimage/morphology/_skeletonize_cy.pyx]: That's true! Thank you @lagru :wink:

File: skimage/morphology/_skeletonize_cy.pyx [hunk 2]
```diff
@@ -8,7 +8,8 @@ cimport numpy as cnp
 cnp.import_array()
 
 
-def _fast_skeletonize(image):
+# def _fast_skeletonize(cnp.uint8_t [:, ::1] image):
+def _fast_skeletonize(cnp.uint8_t [:, ::1] image):
```
Comment (grlee77) [inline skimage/morphology/_skeletonize_cy.pyx:12]:
```suggestion
def _fast_skeletonize(cnp.uint8_t [:, ::1] image):
```

**Review by grlee77 — APPROVED** [review]
Thanks, @rfezzani. I marked on stray comment, but otherwise this looks good

**Comment by rfezzani** [comment]
Thank you all for your reviews :wink:

**Comment by lagru** [comment]
Just wondering, @grlee77 did you intentionally merge without squashing?

**Comment by grlee77** [comment]
No, that was a mistake. I also usually clean up the commit messages a bit.

**Pull request merged**